src/modules/expression/literal/mod.rs

```diff
@@ -11,15 +11,33 @@ pub mod array;
 pub mod range;
 pub mod status;
 
+fn is_escaped(word: &str, symbol: char) -> bool {
+    let mut backslash_count = 0;
+
+    if !word.ends_with(symbol) {
+        return false;
+    }
+
+    for letter in word.chars().rev().skip(1) {
```

Comment:
Just to be annoying XD.

Iterate over indexes from end to start. You won't need to do expensive `rev()`

Reply:
`rev()` does not reverse the elements in array - it simply changes the direction of `DoubleEndedIterator` (which means that if we are in the beginning and we move back, then we move to the end of the collection) to go backwards. It's a very lightweight operation that is advised for this use case.

To learn more you can [see this discussion](https://github.com/rust-lang/rust/issues/11770) or check how the `rev()` is [implemented under the hood](https://github.com/rust-lang/rust/blob/master/library/core/src/iter/adapters/rev.rs).